Snippets/Snippets_6/UpgradeGuides/5to6/CustomErrorHandlingBehaviour.cs

```diff
@@ -0,0 +1,59 @@
+namespace Snippets6.UpgradeGuides._5to6
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus;
+    using NServiceBus.Pipeline;
+
+    #region 5to6-customerrorhandlingbehaviour
+    class CustomErrorHandlingBehaviour : Behavior<ITransportReceiveContext>
+    {
+        public override async Task Invoke(ITransportReceiveContext context, Func<Task> next)
+        {
+            try
+            {
+                await next().ConfigureAwait(false);
+            }
+            catch (MessageDeserializationException)
```

Comment:
It's an example of how to swallow an exception
On Fri, 11 Mar 2016 at 6:36 PM, Mike Minutillo <notifications@github.com>
wrote:

> In Snippets/Snippets_6/UpgradeGuides/5to6/CustomErrorHandlingBehaviour.cs
> <https://github.com/Particular/docs.particular.net/pull/1300#discussion_r55798017>
> :
>
> > +{
> > +    using System;
> > +    using System.Threading.Tasks;
> > +    using NServiceBus;
> > +    using NServiceBus.Pipeline;
> > +
> > +    #region 5to6-customerrorhandlingbehaviour
> > +    class CustomErrorHandlingBehaviour : Behavior<ITransportReceiveContext>
> > +    {
> > +        public override async Task Invoke(ITransportReceiveContext context, Func<Task> next)
> > +        {
> > +            try
> > +            {
> > +                await next().ConfigureAwait(false);
> > +            }
> > +            catch (MessageDeserializationException)
>
> Why does the snippet code swallow this exception?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/docs.particular.net/pull/1300/files#r55798017>
> .
>